mogp_gpu/src/kernel.cu

```diff
@@ -0,0 +1,408 @@
+#include "kernel.hpp"
+#include <stdio.h>
+#include <iostream>
+
+/// Device functions, and global functions (also, confusingly called "kernels") preceded by
```

Comment:
Fair point that "kernel" is overloaded, particularly in this context. "Covariance Function" is probably a more accurate term to use, but I'm not sure it's worth changing everything because I chose some other term back in the day for this.